client/src/main/java/io/pravega/client/stream/impl/TransactionalEventStreamWriterImpl.java

```diff
@@ -145,15 +144,15 @@ public void abort() {
                         log.debug("Got exception while writing to transaction on abort: {}", e.getMessage());
                     }
                 }
-                getAndHandleExceptions(controller.abortTransaction(stream, txId), RuntimeException::new);
+                Futures.getThrowingException(controller.abortTransaction(stream, txId));
```

Comment:
Observation : The abort Transaction operations on the controller are idempotent, hence calling an abortTransaction() successively will not throw an error.(TnxFailedException or any other error).  
The controller client will throw a TxnFailedException if the txn id is not found by the controller during abort().

Reply:
Yes Sandeep